test/react_test_utils_test.dart

```diff
@@ -91,58 +96,101 @@ testUtils({isComponent2: false, dynamic eventComponent, dynamic sampleComponent,
-      group('dragEnd', () => testEvent(Simulate.dragEnd, 'dragEnd'));
-      group('dragEnter', () => testEvent(Simulate.dragEnter, 'dragEnter'));
-      group('dragExit', () => testEvent(Simulate.dragExit, 'dragExit'));
-      group('dragLeave', () => testEvent(Simulate.dragLeave, 'dragLeave'));
-      group('dragOver', () => testEvent(Simulate.dragOver, 'dragOver'));
-      group('dragStart', () => testEvent(Simulate.dragStart, 'dragStart'));
-      group('drop', () => testEvent(Simulate.drop, 'drop'));
-      group('focus', () => testEvent(Simulate.focus, 'focus'));
-      group('gotPointerCapture', () => testEvent(Simulate.gotPointerCapture, 'gotPointerCapture'));
-      group('input', () => testEvent(Simulate.input, 'input'));
-      group('keyDown', () => testEvent(Simulate.keyDown, 'keyDown'));
-      group('keyPress', () => testEvent(Simulate.keyPress, 'keyPress'));
-      group('keyUp', () => testEvent(Simulate.keyUp, 'keyUp'));
-      group('lostPointerCapture', () => testEvent(Simulate.lostPointerCapture, 'lostPointerCapture'));
-      group('mouseDown', () => testEvent(Simulate.mouseDown, 'mouseDown'));
-      group('mouseMove', () => testEvent(Simulate.mouseMove, 'mouseMove'));
-      group('mouseOut', () => testEvent(Simulate.mouseOut, 'mouseOut'));
-      group('mouseOver', () => testEvent(Simulate.mouseOver, 'mouseOver'));
-      group('mouseUp', () => testEvent(Simulate.mouseUp, 'mouseUp'));
-      group('paste', () => testEvent(Simulate.paste, 'paste'));
-      group('pointerCancel', () => testEvent(Simulate.pointerCancel, 'pointerCancel'));
-      group('pointerDown', () => testEvent(Simulate.pointerDown, 'pointerDown'));
-      group('pointerEnter', () => testEvent(Simulate.pointerEnter, 'pointerEnter'));
-      group('pointerLeave', () => testEvent(Simulate.pointerLeave, 'pointerLeave'));
-      group('pointerMove', () => testEvent(Simulate.pointerMove, 'pointerMove'));
-      group('pointerOver', () => testEvent(Simulate.pointerOver, 'pointerOver'));
-      group('pointerOut', () => testEvent(Simulate.pointerOut, 'pointerOut'));
-      group('pointerUp', () => testEvent(Simulate.pointerUp, 'pointerUp'));
-      group('scroll', () => testEvent(Simulate.scroll, 'scroll'));
-      group('submit', () => testEvent(Simulate.submit, 'submit'));
-      group('touchCancel', () => testEvent(Simulate.touchCancel, 'touchCancel'));
-      group('touchEnd', () => testEvent(Simulate.touchEnd, 'touchEnd'));
-      group('touchMove', () => testEvent(Simulate.touchMove, 'touchMove'));
-      group('touchStart', () => testEvent(Simulate.touchStart, 'touchStart'));
-      group('transitionEnd', () => testEvent(Simulate.transitionEnd, 'transitionEnd'));
-      group('wheel', () => testEvent(Simulate.wheel, 'wheel'));
+      void Function(SyntheticEvent) _expectEventType<T>() {
+        return (SyntheticEvent event) {
+          expect(event.isClipboardEvent, T == SyntheticClipboardEvent);
```

Comment:
I like this solution, but I'm not sure that type-equality between reified type parameters is guaranteed by the language, so I'm hesitant to rely on it without asserting that exactly one of these conditions is `true`.

For instance, all of them could be false if the type isn't exactly equal, or they could all be true if JS interop types end up being the same in dart2js or something weird and they're equal to each other.

Instead of generic parameters, could we use enums from `event_helpers_test.dart` instead?

Reply:
Updated in 03454a1